TMinus/Launch.swift

```diff
@@ -29,6 +30,31 @@ struct Launch {
     //MARK: Private
 }
 
+extension Launch: NotificationType {
+    
+    func makeNotificationRequest() -> UNNotificationRequest {
+        let date = startDate.addingTimeInterval(-.oneHour)
+        let components = Calendar.current.dateComponents([.year, .month, .day, .hour, .minute, .second], from: date)
+        
+        let content = UNMutableNotificationContent()
+        var title = rocketName
+        if let missionName = missionName, !missionName.isEmpty {
+            title = "\(title) - \(missionName)"
+        }
+        content.title = title
+        content.body = "The launch window is scheduled to open in one hour."
+        
+        let trigger = UNCalendarNotificationTrigger(dateMatching: components, repeats: false)
+        let identifier = launchIdentifier()
+        return UNNotificationRequest(identifier: identifier, content: content, trigger: trigger)
+    }
+    
+    private func launchIdentifier() -> String {
```

Comment:
Another small stylistic thing, but I think a "Swiftier" approach is to make this a generated get-only property. I usually go this route when there's no parameters or real calculations involved.